db/database.go

```diff
@@ -308,6 +268,49 @@ func NewDatabaseContext(dbName string, bucket base.Bucket, autoImport bool, opti
 		}
 	}
 
+	context.sequences, err = newSequenceAllocator(bucket)
+	if err != nil {
+		return nil, err
+	}
+	lastSeq, err := context.sequences.lastSequence()
+	if err != nil {
+		return nil, err
+	}
+
+	if options.IndexOptions == nil {
+		// In-memory channel cache
+		context.SequenceType = IntSequenceType
+		context.changeCache = &changeCache{}
+	} else {
+		// KV channel index
+		context.SequenceType = ClockSequenceType
+		context.changeCache = &kvChangeIndex{}
+		context.SequenceHasher, err = NewSequenceHasher(options.SequenceHashOptions)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Callback that is invoked whenever a set of channels is changed in the ChangeCache
+	onChange := func(changedChannels base.Set) {
+		context.mutationListener.Notify(changedChannels)
+	}
+
+	// Initialize the ChangeCache
+	context.changeCache.Init(
+		context,
+		SequenceID{Seq: lastSeq},
+		onChange,
+		options.CacheOptions,
+		options.IndexOptions,
+	)
+
+	// Set the DB Context onChange callback to call back the changecache DocChanged callback
+	context.SetOnChangeCallback(context.changeCache.DocChanged)
```

Comment:
Initializing the channel cache (in particular setting the OnChange callback) after we start the DCP feed has the potential to miss DCP events that start arriving as soon as the feed is started, but before we hit this code.

That's definitely a problem in scenarios we need to see all mutations (import, migrate).

To safely cover all cases, I think we need the changeCache to lazy initialize lastSeq the first time changeCache.DocChanged is called.